x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx

```diff
@@ -0,0 +1,472 @@
+
+      if (timeRange.useSubAgg) {
+        const categoriesInBucket = categorizationResult.categories
+          .map((category) => ({
+            ...category,
+            count: category.subFieldCount ?? category.subTimeRangeCount!,
+            examples: category.subFieldExamples!,
+            sparkline: category.subFieldSparkline,
+          }))
+          .filter((category) => category.count > 0)
+          .sort((a, b) => b.count - a.count);
+        setData({
+          categories: categoriesInBucket,
+          displayExamples: hasExamples,
+          totalCategories: categories.length,
+        });
+      } else {
+        setData({
+          categories,
+          displayExamples: hasExamples,
+          totalCategories: categories.length,
+        });
+      }
+    } catch (error) {
+      if (error.name === 'AbortError') {
+        // ignore error
+      } else {
+        toasts.addError(error, {
+          title: i18n.translate('xpack.aiops.logCategorization.errorLoadingCategories', {
+            defaultMessage: 'Error loading categories',
+          }),
+        });
+      }
+    }
+
+    if (mounted.current === true) {
+      setLoading(false);
+    }
+  }, [
+    dataView,
+    loading,
+    selectedField,
+    earliest,
+    latest,
+    minimumTimeRangeOption,
+    cancelRequest,
+    getMinimumTimeRange,
+    searchQuery,
+    runValidateFieldRequest,
+    embeddingOrigin,
+    runCategorizeRequest,
+    intervalMs,
+    toasts,
+  ]);
+
+  useEffect(
+    function triggerAnalysis() {
+      const buckets = documentStats.documentCountStats?.buckets;
+      if (buckets === undefined || currentDocumentStatsHash === null) {
+        return;
+      }
+
+      if (
+        currentDocumentStatsHash !== previousDocumentStatsHash ||
+        (currentAdditionalConfigsHash !== previousAdditionalConfigsHash &&
+          currentDocumentStatsHash !== null)
+      ) {
+        randomSampler.setDocCount(documentStats.totalCount);
+        setEventRate(
+          Object.entries(buckets).map(([key, docCount]) => ({
+            key: +key,
+            docCount,
+          }))
+        );
+        loadCategories();
+        setPreviousDocumentStatsHash(currentDocumentStatsHash);
+        setPreviousAdditionalConfigsHash(currentAdditionalConfigsHash);
+      }
+    },
+    [
+      loadCategories,
+      randomSampler,
+      previousDocumentStatsHash,
+      fieldValidationResult,
+      currentDocumentStatsHash,
+      currentAdditionalConfigsHash,
+      documentStats.documentCountStats?.buckets,
+      documentStats.totalCount,
+      previousAdditionalConfigsHash,
+    ]
+  );
+
+  useEffect(
+    function refreshTriggeredFromButton() {
+      if (input.lastReloadRequestTime !== undefined) {
+        setPreviousDocumentStatsHash(0);
+        setPreviousAdditionalConfigsHash(null);
+        forceRefresh();
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [input.lastReloadRequestTime]
```

Comment:
can we list the complete set of deps?

Reply:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846